src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionPal.Windows.cs

```diff
@@ -409,17 +411,14 @@ public void RegisterForCancellation(CancellationToken cancellationToken)
                         var @this = (GetAddrInfoExState)o!;
                         int cancelResult = 0;
 
-                        lock (@this)
```

Comment:
Is there a use-after-free race condition here with `context` being freed at the end of `ProcessResult`?

Reply:
Good catch, I overlooked this race. Looks like we should implement a more intrusive change and add some sort of ref-counting here.

Reply:
a2a9355